PhisicsInterface.py

```diff
@@ -0,0 +1,293 @@
+    """
+      This method is used to retrieve the command (in tuple format) needed to launch the Code.
+      See base class.  Collects all the clargs and the executable to produce the command-line call.
+      Returns tuple of commands and base file name for run.
+      Commands are a list of tuples, indicating parallel/serial and the execution command to use.
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path (e.g. /home/path_to_executable/code.exe)
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ In, fargs, dict, optional, a dictionary containing the axuiliary input file variables the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 aux0extension =0 .aux0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
+    """
+    if self.mrtauStandAlone == True: 
+      executable = self.switchExecutable()
+    found = False
+    index = 0
+    outputfile = 'out~'+inputFiles[index].getBase()
+    #printprint (outputfile)
+    if clargs: addflags = clargs['text']
+    else     : addflags = ''
+    #commandToRun = executable + ' -i ' + inputFiles[index].getFilename() + ' -o ' + outputfile  + '.o' + ' -r ' + outputfile  + '.r' + addflags
+    #commandToRun = executable + ' -i ' + inputFiles[index].getFilename() + ' -o ' + outputfile  + '.o' +  addflags
+    commandToRun = executable
+    commandToRun = commandToRun.replace("\n"," ")
+    commandToRun  = re.sub("\s\s+" , " ", commandToRun )
+    #print (commandToRun)
+    returnCommand = [('parallel',commandToRun)], outputfile
+    #print (commandToRun)
+    #print (returnCommand)
+    return returnCommand
+  
+  def finalizeCodeOutput(self,command,output,workingDir):
+    """
+      This method is called by the RAVEN code at the end of each run (if the method is present, since it is optional).
+      It can be used for those codes, that do not create CSV files to convert the whatever output format into a csv
+      @ In, command, string, the command used to run the just ended job
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, output, string, optional, present in case the root of the output file gets changed in this method.
+    """
+    output = 'Dpl_INSTANT.outp-0'
+    splitWorkDir = workingDir.split('/')
+    pertNumber = splitWorkDir[-1]
+    outputobj=phisicsdata.phisicsdata(output, workingDir, self.mrtauStandAlone, self.jobTitle)
+    return "keff"+str(pertNumber).strip()
```

Comment:
probably the naming here can be more generic....I do not thing you need to modify the name of the ```output``` root

Reply:
and you do not need to add the ```pertNumber``` since each output is in a separate folder (e.g. ```./1/```, ```./2/```, etc.)